.kres.yaml

```diff
@@ -0,0 +1,634 @@
+---
+kind: golang.Toolchain
+spec:
+  # image: ghcr.io/siderolabs/tools:v1.8.0-alpha.0
+  buildTags:
+    - tcell_minimal
+    - grpcnotrace
+  makefile:
+    extraVariables:
+      - name: NAME
```

Comment:
I think this is a bit of a disaster, we're basically rewriting `Makefile` as `.kres.yaml`, 

Can we take a step back and try to analyze what can be done better if we can do that? Or at least analyze some options?